src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -80,6 +81,22 @@ public void EndIntervalMeasure(string propertyName)
             _intervalList.Add(new Tuple<string, TimeSpan>(propertyName, _intervalWatch.Elapsed));
         }
 
+        /// <summary> Start interval measure. </summary>
+        /// <param name="stopWatch"> Stopwatch used to measure the time interval. </param>
+        public void StartIntervalMeasure(Stopwatch stopWatch)
+        {
+            stopWatch.Restart();
+        }
+
+        /// <summary> End interval measure. </summary>
+        /// <param name="propertyName"> Property name to represents the interval. </param>
+        /// <param name="stopWatch"> Stopwatch used to measure the time interval. </param>
+        public void EndIntervalMeasure(string propertyName, Stopwatch stopWatch)
+        {
+            stopWatch.Stop();
```

Comment:
The `TelemetryActivity` class didn't start the stopwatch passed in, so I don't think it should stop it. It's not the "owner" of the stopwatch.